### PR TITLE
Fix #37451. Adds tests for invalid pseudo-element

### LINKED ISSUE #37451: Missing test for real pseudo name without `:` or `::`
https://github.com/web-platform-tests/wpt/blob/4a4653e088039aec186d6dc1d488120d77695c3a/css/cssom/getComputedStyle-pseudo.html#L139-L152

### PR DESCRIPTION
**Pull request by karlcow**
See #37451

### CODE REVIEW

**Comment by karlcow** [comment]
@nt1m For https://bugs.webkit.org/show_bug.cgi?id=243539
and https://github.com/WebKit/WebKit/pull/7468

Probably @Emilio too as it was his test.

File: css/cssom/getComputedStyle-pseudo.html [hunk 1]
```diff
@@ -150,4 +158,19 @@
...
+  assert_equals(markerStyle.color, "rgb(0, 128, 0)", "color is green.");
+}, `getComputedStyle() for ::marker returns the right value`);
+test(() => {
+  let li = document.querySelector('li');
+  let markerStyle = getComputedStyle(li, 'marker');
+  assert_not_equals(markerStyle.color, "rgb(0, 128, 0)", "no value is returned.");
+}, `getComputedStyle() for invalid marker should not return the color of marker`);
+test(() => {
+  let li = document.querySelector('li');
+  let markerStyle = getComputedStyle(li, 'marker');
```
Comment (lilles) [inline css/cssom/getComputedStyle-pseudo.html:173]: This test is doing the exact same query as the previous test, why should they be different?

File: css/cssom/getComputedStyle-pseudo.html [hunk 2]
```diff
@@ -150,4 +158,19 @@
     getComputedStyle(div, ":totallynotapseudo").length == 0,
     "Should return an empty style for unknown pseudo-elements starting with colon");
 }, "Unknown pseudo-elements");
+test(() => {
+  let li = document.querySelector('li');
+  let markerStyle = getComputedStyle(li, ':marker');
```
Comment (lilles) [inline css/cssom/getComputedStyle-pseudo.html:163]: Why is there a single colon, not a double?

Is the intention of the test to make "::marker" a valid query but make ":marker" and "marker" invalid ones?

Comment (emilio) [reply css/cssom/getComputedStyle-pseudo.html:163]: This should _not_ return a style, `::marker` is not one of the [CSS2 pseudo-elements](https://drafts.csswg.org/selectors-4/#pseudo-element-syntax) (`:before/:after/:first-line/:first-letter`), so it should be treated as any other invalid pseudo-element would.

File: css/cssom/getComputedStyle-pseudo.html (hunk 1 above)
Comment (emilio) [reply css/cssom/getComputedStyle-pseudo.html:173]: Yeah this doesn't make sense. Per [spec](https://drafts.csswg.org/cssom/#dom-window-getcomputedstyle) all of these should return an empty style.

**Review by emilio — CHANGES_REQUESTED** [review]

File: css/cssom/getComputedStyle-pseudo.html (hunk 2 above)
Comment (karlcow) [reply css/cssom/getComputedStyle-pseudo.html:163]: This is a typo from me. I meant to write `::marker` and yes `:marker` and `marker` are invalid.

File: css/cssom/getComputedStyle-pseudo.html (hunk 1 above)
Comment (karlcow) [reply css/cssom/getComputedStyle-pseudo.html:173]: And my markup and style above is also wrong. 🤦 End of the day coding. 
I failed to explain my intent. My intent is to specifically make sure that `:marker` and `marker` do not trigger the color.

This is what is happening currently.

```html
<style>
#pseudo-invalid::marker {
  color: rgb(0 128 0);
}
#pseudo-invalid {
  color: rgb(255 0 0)
}
</style>
<!-- dot is green, text is red -->
  <ul><li id="pseudo-invalid">Item</li></ul>
```

```js
window.getComputedStyle(document.querySelector('#pseudo-invalid'), 'marker').color 
window.getComputedStyle(document.querySelector('#pseudo-invalid'), ':marker').color 
window.getComputedStyle(document.querySelector('#pseudo-invalid'), '::marker').color 
window.getComputedStyle(document.querySelector('#pseudo-invalid'), 'foobar').color
window.getComputedStyle(document.querySelector('#pseudo-invalid'), ':foobar').color
window.getComputedStyle(document.querySelector('#pseudo-invalid'), '::foobar').color
```

Safari Tech Preview 159

* `marker`: `rgb(0, 128, 0)` aka dot color (WRONG)
* `:marker`: `rgb(0, 128, 0)` aka dot color (WRONG) 
* `::marker`: `rgb(0, 128, 0)` aka dot color (RIGHT)
* `foobar`: `rgb(255, 0, 0)` aka `li` color 
* `:foobar`: `rgb(255, 0, 0)` aka `li` color 
* `::foobar`: `rgb(255, 0, 0)` aka `li` color 

Firefox Nightly 109.0a1 (2022-12-08)

* `marker`: `rgb(255, 0, 0)` aka `li` color 
* `:marker`: `` nothing
* `::marker`: `rgb(0, 128, 0)` aka dot color (RIGHT)
* `foobar`: `rgb(255, 0, 0)` aka `li` color 
* `:foobar`: `` nothing
* `::foobar`: `` nothing

Chrome Canary 110.0.5462.2

* `marker`: `rgb(0, 128, 0)` aka dot color (WRONG)
* `:marker`: `rgb(0, 128, 0)` aka dot color (WRONG) 
* `::marker`: `rgb(0, 128, 0)` aka dot color (RIGHT)
* `foobar`: `rgb(255, 0, 0)` aka `li` color 
* `:foobar`: `rgb(255, 0, 0)` aka `li` color 
* `::foobar`: `rgb(255, 0, 0)` aka `li` color 

So basically I want to make sure that when there is a `::pseudoId`, that `:pseudoId` OR `pseudoId` is not interpreted as known when the string exist among pseudo-elements.

So Safari Tech Preview and Chrome Canary are clearly wrong. 
And it's why I want to make sure in the tests.

And there's also a confusing logic for the fallback on either empty style or `li` style for wrong `pseudoId`

@Emilio does it clarify a bit. In any case my file here is fully wrong. 
Do you need Is it worth to add to the test the case where the pseudoId string of the pseudo element is a known one, such as in `marker` and `:marker`.

File: css/cssom/getComputedStyle-pseudo.html (hunk 2 above)
Comment (karlcow) [reply css/cssom/getComputedStyle-pseudo.html]: I fixed my typos and added two tests. The goal here is to make sure that a known `pseudoId` such as `marker` in `marker` or `:marker` is not considered valid for WebKit and Blink.  I hope it clarifies the intent.

**Review by emilio — APPROVED** [review]
Maybe check that `::marker` does work? But yeah these look better :)

**Comment by karlcow** [comment]
> Maybe check that `::marker` does work? But yeah these look better :)

I can do that :)

File: css/cssom/getComputedStyle-pseudo.html (hunk 1 above)
Comment (karlcow) [reply css/cssom/getComputedStyle-pseudo.html]: Another weird thing.

@Emilio Is it a bug from Firefox?  `:marker` vs `marker` or even `:foobar` vs `foobar`

```
Another difference.

data:text/html,<li style="color:red">item</li>

window.getComputedStyle(document.querySelector('li'), "::marker").color

Safari:  'rgb(255, 0, 0)'
Firefox: 'rgb(255, 0, 0)'
Chrome:  'rgb(255, 0, 0)' 

window.getComputedStyle(document.querySelector('li'), ":marker").color

Safari:  'rgb(255, 0, 0)'
Firefox: ''
Chrome:  'rgb(255, 0, 0)' 

window.getComputedStyle(document.querySelector('li'), "marker").color

Safari:  'rgb(255, 0, 0)'
Firefox: 'rgb(255, 0, 0)'
Chrome:  'rgb(255, 0, 0)' 

```

is it expected that `marker` or `foobar` returns a color but that `:pseudoId` returns empty string? That doesn't seem consistent.

Comment (emilio) [reply css/cssom/getComputedStyle-pseudo.html]: That follows the spec as far as I can tell:

> If pseudoElt is provided, is not the empty string, and starts with a colon, then:

Comment (emilio) [reply css/cssom/getComputedStyle-pseudo.html]: IIRC we needed the "starts with a colon check" due to backwards compat shenanigans (e.g. people would do `getComputedStyle(foo, "null")`, but I might be misremembering

Comment (karlcow) [reply css/cssom/getComputedStyle-pseudo.html]: ok. Thanks, there are more tests needed and fixed one current.
I will update the PR.

**Comment by dbaron** [comment]
Did you intend to merge this?

**Comment by karlcow** [comment]
> Did you intend to merge this?

@dbaron I need to finish the testing on my side. To make it extra-sure. Almost there. Thanks for the bell.

**Comment by karlcow** [comment]
@dbaron sorry for the long… super long… delay. This is now finished. 
And can be merged.
